### DISCUSSION TITLE
List notes and Watchlist/Recommendations description updating

### DISCUSSION THREAD

**Discussion by IrishStevo** [trakt/trakt-api#401] [Q&A] [answered]
I have a couple of queries about lists. I'm creating this as a discussion in case I'm missing something but happy to create issues for each of these if needed.

1. Is it possible to update notes on list items? I can't find an endpoint that looks like it will allow for that. The only way I have found so far is to delete all the list items and re-add them to the list with the updated notes, which is a little destructive and prevents supporting Recently Added sorting.

2. Is it possible to update the description of the Watchlist and Recommendations lists? Again, I can't find an endpoint that looks like it would allow this.

**Comment by rectifyer** [comment, marked as answer]
I'll have an update out tomorrow to allow updating notes on list items. API docs are updated so you can see how it will work.

https://trakt.docs.apiary.io/#reference/users/update-list-item/update-a-list-item
https://trakt.docs.apiary.io/#reference/sync/update-watchlist-item/update-a-watchlist-item
https://trakt.docs.apiary.io/#reference/sync/update-personal-recommendation-item-beta/update-a-personal-recommendation-item

**Reply by IrishStevo** [reply]
That's great, thanks!

Would you consider also adding an endpoint that allows bulk updating multiple item notes in one API call at some point?

My use case involves regularly updating notes for all items in multiple lists, with nearly all lists having 100 or more items. Making individual calls for each list item while sticking to the 1 call per second for writes would be a little painful.

**Reply by rectifyer** [reply]
Good point on the rate limiting. I will look into this more. The main reason I didn't make a bulk method was to make it a fast call. If you are updating 500 items on a list the method would just take a while to complete.

**Comment by rectifyer** [comment]
We don't have methods to update the watchlist and recommendations lists.

**Reply by IrishStevo** [reply]
Would you also consider adding endpoints for this at some point in the future?

**Reply by rectifyer** [reply]
Yes, I have this on my task list to look into.

**Reply by rectifyer** [reply]
Watchlist and Favorites allow updating the description and sorting, like the website.

https://trakt.docs.apiary.io/#reference/sync/update-watchlist/update-watchlist
https://trakt.docs.apiary.io/#reference/sync/update-favorites/update-favorites

**Reply by IrishStevo** [reply]
Both endpoints work great. Thanks!
